### DISCUSSION TITLE
关于大型模式 Pretouch()

### DISCUSSION THREAD

**Discussion by npnpwqf** [bytedance/sonic#434] [General]
请教：多重嵌套结构体，（nesting depth > option.DefaultMaxInlineDepth），option.WithCompileRecursiveDepth(loop)，此 loop 值推荐设置多少，或者该数据根据什么来定。

**Comment by AsterDY** [comment]
MaxInlineDepth 按默认值就好，CompileRecursiveDepth 影响预热编译时间，如果想完全编完的话可以设置为  **(CompileRecursiveDepth+1) * MaxlnlineDepth > nesting depth**； 此外，如果结构体中存在 interface{} 类型字段会导致 pretouch 无法继续进行（因为并不知道其运行时类型），这种情况下建议直接在 init 阶段使用 **线上真实数据&&结构体 Marshal()&&Unmarshal() 一次**
